src/mine_loop.rs

```diff
@@ -427,23 +430,27 @@ pub(crate) async fn make_coinbase_transaction_stateless(
 /// coinbase amount, since the guesser fee fraction is guaranteed to be in the
 /// range \[0;1\].
 ///
-/// Returns an array of two TxOutputs: liquid first, timelocked second.
+/// Returns: Either the empty list, or two outputs, one immediately liquid, the
+/// other timelocked for three years, as required by the consensus rules.
+///
+/// # Panics
+///
+/// If the provided guesser fee fraction is not between 0 and 1 (inclusive).
 pub(crate) fn composer_outputs(
     coinbase_amount: NativeCurrencyAmount,
     composer_parameters: ComposerParameters,
     timestamp: Timestamp,
-) -> Result<[TxOutput; 2]> {
+) -> TxOutputList {
     let guesser_fee =
         coinbase_amount.lossy_f64_fraction_mul(composer_parameters.guesser_fee_fraction());
 
-    let Some(amount_to_composer) = coinbase_amount.checked_sub(&guesser_fee) else {
-        bail!(
-            "Guesser fee may not exceed coinbase amount. coinbase_amount: {}; guesser_fee: {}; guesser fee fraction: {}.",
-            coinbase_amount.to_nau(),
-            guesser_fee.to_nau(),
-            composer_parameters.guesser_fee_fraction()
-        );
-    };
+    let amount_to_composer = coinbase_amount
+        .checked_sub(&guesser_fee)
+        .expect("total_composer_fee cannot exceed coinbase_amount");
```

Comment:
1. previously this was returning an error.  now it panics.  why?
2. the expect message seems wrong.  I think it should be: "guesser_fee cannot exceed coinbase_amount".  which also matches the pre-existing error message.

Reply:
That `bail!` was dead code -- could never be reached because of the above expression ` coinbase_amount.lossy_f64_fraction_mul(composer_parameters.guesser_fee_fraction());`

Reply:
You're right about the wrong error message. Sorry about that.